src/include/cell.i

```diff
@@ -369,11 +487,19 @@ __wt_cell_pack_value_match(WT_CELL *page_cell,
 		++a;
 	} else if (WT_CELL_TYPE(a[0]) == WT_CELL_VALUE) {
 		rle = (a[0] & WT_CELL_64V) != 0;
-		ts = (a[0] & WT_CELL_TIMESTAMPS) != 0;
+		validity = (a[0] & WT_CELL_SECOND_DESC) != 0;
 		++a;
-		if (ts) {
-			WT_RET(__wt_vunpack_uint(&a, 0, &v));	/* Skip TS */
-			WT_RET(__wt_vunpack_uint(&a, 0, &v));
+		if (validity) {			/* Skip validity window */
```

Comment:
Having to skip like this makes me a bit sad - though I can't think of a better solution. We can't really move where the validity window sits in the data structure, and we don't want to pay the cost of saving the size of the validity window into the cell.

Reply:
> Having to skip like this makes me a bit sad - though I can't think of a better solution. We can't really move where the validity window sits in the data structure, and we don't want to pay the cost of saving the size of the validity window into the cell.

We could move where the validity window sits in the cell. I started out putting it at the end of the cell after the data chunk because that was an easier code change, but .eventually decided that was harder to think about, so I went with what's there now. If you feel strongly, this could change.

Reply:
Overall I think it's better where you have added it.